### DISCUSSION TITLE
Markdown Extensions - Markwhen

### DISCUSSION THREAD

**Discussion by mwx23** [marimo-team/marimo#1890] [Q&A]
Hi,

I've recently come across your fantastic project and also the wonderful Markdown Extension [Markwhen](https://github.com/mark-when/markwhen).

I'd like to help and contribute a plugin for Markwhen to Marimo, however I'm a bit stuck as where to start. 

I presume it would be similar to the Mermaid plugin so replicating `marimo._plugins.stateless.mermaid` and `frontend/src/plugins/layout/mermaid` does that sound like the right approach?

I'm happy to submit a pull request when I'm done, or is there a planned mechanism for third party plugin extensions e.g. `pip install marimo-markwhen-extension` and have Marimo load them.

Thanks!

**Comment by mscolnick** [comment]
hi @mwx23 - this sounds like a great contribution. you may definitely submit a pull request and i'd be happy to review. 

`mermaid` plugin would be a great place to start. one thing i noticed with the online version is that it is by-directional. not sure if you were going to do something similar, but our UI plugins cannot update python source code yet (but that is an API point we could maybe add)
